### PR TITLE
Song GO client updates

### PR DESCRIPTION
**Pull request by leoraba**
Fix GO client errors and update to version 1.0.0

### CODE REVIEW

File: song-go-client/cmd/search.go [hunk 1]
```diff
@@ -21,9 +21,10 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"strings"
+
```
Comment (headsoup) [inline song-go-client/cmd/search.go:25]: this space is just to distinguish from the urls, or?

File: Makefile [hunk 2]
```diff
@@ -120,7 +120,7 @@ help:
 	@echo "To dry-execute a target run: make -n <target> "
 	@echo
 	@echo "Available Targets: "
-	@grep '^[A-Za-z][A-Za-z0-9_-]\+:.*' $(ROOT_DIR)/Makefile | sed 's/:.*//' | sed 's/^/\t/'
+	@grep '^[A-Za-z0-9_-]\+:.*' $(ROOT_DIR)/Makefile | sed 's/:.*//' | sed 's/^/\t/'
```
Comment (leoraba) [inline Makefile:123]: print all available targets starting with any letter, number or symbol

File: song-go-client/cmd/status.go [hunk 3]
```diff
@@ -19,40 +19,28 @@ package cmd
...
 	fmt.Println(responseBody)
 }
 
 var statusCmd = &cobra.Command{
-	Use:   "status -p OR status <uploadID>",
-	Short: "Get status of uploaded analysis",
-	Long:  `Get status of uploaded analysis`,
+	Use:   "status",
+	Short: "Get Song Server Status",
+	Long:  `Get Song Server Status`,
```
Comment (headsoup) [inline song-go-client/cmd/status.go:42]: 👍 concise

File: song-go-client/cmd/version.go [hunk 4]
```diff
@@ -32,6 +32,6 @@ var versionCmd = &cobra.Command{
 	Short: "Version number",
 	Long:  `Prints the version of this SONG CLI Application`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("v0.0.1 Alpha")
+		fmt.Println("v1.0.0")
```
Comment (headsoup) [inline song-go-client/cmd/version.go:35]: 💪

File: song-go-client/cmd/submit.go [hunk 5]
```diff
@@ -45,16 +43,16 @@ func upload(filePath string) {
...
-	responseBody := client.Upload(studyID, b, asyncFlag)
+	responseBody := client.Submit(studyID, b)
 	fmt.Println(string(responseBody))
 }
 
-var uploadCmd = &cobra.Command{
-	Use:   "upload <filename>",
-	Short: "Upload Analysis Metadata",
-	Long:  `Uploads Metadata JSON describing an analysis and files for validation`,
+var submitCmd = &cobra.Command{
```
Comment (leoraba) [inline song-go-client/cmd/submit.go:50]: `upload` SONG operation was replaced by `submit`

File: song-go-client/example/exampleVariantCall.json [hunk 6]
```diff
@@ -1,76 +1,81 @@
...
-    "info" : {
-        "randomField16" : "alternatively, put some extra EXPERIMENT fields here"
+    "randomField14": "we can define any EXPERIMENT field. For example, randomField14",
+    "randomField15": "as a second example, we can define another random EXPERIMENT field called randomField15",
+    "info": {
+      "randomField16": "alternatively, put some extra EXPERIMENT fields here"
     }
   },
-  "sample": [
+  "samples": [
```
Comment (leoraba) [inline song-go-client/example/exampleVariantCall.json:16]: transform example JSON file accordingly to latest schema version

File: song-go-client/cmd/search.go (hunk 1 above)
Comment (leoraba) [reply song-go-client/cmd/search.go:25]: correct. but it's prettier adding the space.

File: song-go-client/song/endpoint.go [hunk 7]
```diff
@@ -58,15 +49,6 @@ func truth(condition bool) string {
 	return "false"
 }
 
-// Save saves the specified uploadID as an analysis assuming it had passed validation
-func (s *Endpoint) Save(studyID string, uploadID string, ignoreCollisions bool) url.URL {
```
Comment (headsoup) [inline song-go-client/song/endpoint.go:62]: you mentioned `upload` being replaced... also `save`?

Comment (leoraba) [reply song-go-client/song/endpoint.go:62]: Right. The SONG `submit` endpoint merges the `upload` and `save`.

Comment (headsoup) [reply song-go-client/song/endpoint.go:62]: hmm I remember that whole epic now. cool!

**Pull request merged**